emailproxy.py

```diff
@@ -337,9 +341,141 @@ def add_account(username):
     @staticmethod
     def save():
         if AppConfig._LOADED:
+            if AppConfig.aws_secrets_accounts():
+                config_to_write = AppConfig.aws_secrets_save()
+            else:
+                config_to_write = AppConfig._PARSER
+
             with open(CONFIG_FILE_PATH, mode='w', encoding='utf-8') as config_output:
-                AppConfig._PARSER.write(config_output)
+                config_to_write.write(config_output)
+
+    @staticmethod
+    def aws_secrets_accounts():
+        return [a for a in AppConfig._ACCOUNTS if 'aws_secret' in AppConfig._PARSER[a]]
+
+    @staticmethod
+    def aws_secrets_ids():
+        return dict.fromkeys([AppConfig._PARSER[account]['aws_secret'] for account in AppConfig.aws_secrets_accounts()], {})
+
+    @staticmethod
+    def aws_secrets_boto3client():
+        try:
+            # Load dependencies and create client
+            global boto3
+            import boto3
+            global botocore
+            import botocore.exceptions
+            aws_client = boto3.client('secretsmanager')
+        except Exception as e:
+            Log.error('Failed to load AWS SDK - "aws_secret" is specified in config file for some accounts, have you installed requirements-aws-secrets.txt?')
+            Log.debug(Log.error_string(e))
+            return
+        else:
+            return aws_client
+
+    @staticmethod
+    def aws_secrets_debuglog_boto3error(err):
+        Log.debug(err.response.get("Error", {}).get("Code") + ":",
+                  err.response.get("Error", {}).get("Message"))
```

Comment:
Why not just `err.response['Error']['Code']` and `err.response['Error']['Message']`?

What is the default value of `{}` for here?

Reply:
Agreed, the convoluted code is a symptom of copy pasting. Will revise to be clearer.